pkg_resources/__init__.py

```diff
@@ -65,20 +65,25 @@
 from os import open as os_open
 from os.path import isdir, split
 
-from pkg_resources.extern.jaraco.text import (
+from pkg_resources._vendor.jaraco.text import (
     yield_lines,
     drop_comment,
     join_continuation,
 )
 
-from pkg_resources.extern import platformdirs
-from pkg_resources.extern import packaging
+from ._vendor import platformdirs
+from ._vendor import packaging
+from ._vendor.packaging.requirements import (
+    # These imports are explicit only to avoid mypy error
+    InvalidRequirement as _packaging_InvalidRequirement,
+    Requirement as _packaging_Requirement,
+)
 
-__import__('pkg_resources.extern.packaging.version')
-__import__('pkg_resources.extern.packaging.specifiers')
-__import__('pkg_resources.extern.packaging.requirements')
-__import__('pkg_resources.extern.packaging.markers')
-__import__('pkg_resources.extern.packaging.utils')
+__import__('pkg_resources._vendor.packaging.version')
+__import__('pkg_resources._vendor.packaging.specifiers')
+__import__('pkg_resources._vendor.packaging.requirements')
+__import__('pkg_resources._vendor.packaging.markers')
+__import__('pkg_resources._vendor.packaging.utils')
```

Comment:
I think this explicit import might render `__import__('pkg_resources._vendor.packaging.requirements')` useless?
On that note I also don't see any `packaging.specifiers` in this module.
`packaging.version` is only used for `Version` and `invalidVersion`.
`packaging.markers` is only used for `Marker` and `InvalidMarker`.
`packaging.utils` is only used for `canonicalize_name`.

Unless I misunderstand what else these imports could be for, they can likely be removed even outside this PR.

Reply:
Since I couldn't do a suggestion on GitHub, here's what it'd look like:
```py
from ._vendor import platformdirs
from ._vendor.packaging.markers import Marker, InvalidMarker
from ._vendor.packaging.requirements import (
    InvalidRequirement as _packaging_InvalidRequirement,
    Requirement as _packaging_Requirement,
)
from ._vendor.packaging.utils import canonicalize_name
from ._vendor.packaging.version import Version, InvalidVersion
```

Reply:
I extracted what I saw as valid changes on their own into https://github.com/pypa/setuptools/pull/4348